astroquery/esa/xmm_newton/core.py

```diff
@@ -273,14 +277,53 @@ def get_columns(self, table_name, *, only_names=True, verbose=False):
                 break
 
         if columns is None:
-            raise ValueError("table name specified is not found in "
-                             "XSA TAP service")
+            raise ValueError("table name specified is not found in XSA TAP service")
 
         if only_names:
             return [c.name for c in columns]
         else:
             return columns
 
+    def _create_link(self, observation_id, **kwargs):
+        link = f"{self.data_aio_url}obsno={observation_id}"
+        link = link + "".join("&{0}={1}".format(key, val)
+                              for key, val in kwargs.items())
+        return link
+
+    def _request_link(self, link, cache):
+        # we can cache this HEAD request - the _download_file one will check
+        # the file size and will never cache
+        response = self._request('HEAD', link, save=False, cache=cache)
+        # Get original extension
+        if 'Content-Type' in response.headers and 'text' not in response.headers['Content-Type']:
+            _, params = cgi.parse_header(response.headers['Content-Disposition'])
+        elif response.status_code == 401:
+            error = "Data protected by proprietary rights. Please check your credentials"
+            raise LoginError(error)
+        elif 'Content-Type' not in response.headers:
+            error = "Incorrect credentials"
+            raise LoginError(error)
+        response.raise_for_status()
+        return params
+
+    def _get_username_and_password(self, credentials_file):
```

Comment:
this should use https://github.com/astropy/astroquery/blob/f18f02bf8c76df061179ec912ca62e9c6e0bfb8b/astroquery/query.py#L475

Reply:
Would it be okay to get an explanation to why this change is needed?

Reply:
to avoid redundant code.  We already have a global tool for authenticating with remote services, so we should reuse that.  Our philosophy in astroquery is that all tools should use a common framework when possible.  This is critical for long-term code maintenance.

The existing tool doesn't use a config file, though, so if you want username/password to be globally configurable, you can add that functionality in `query.py`.  That seems like a useful capability to add.

Relatedly, though, this would allow you to use the already-present configuration system in astroquery rather than setting up a separate config system.  Users could just put their username & pw in the `~/.astropy/config/astroquery.cfg` file under the `[xmm_newton]` section.

Reply:
Thank you very much for pointing this out and have implemented the changes. Do you wish for me to put the _get_username_and_password into query.py so we can use a config file?

Reply:
No, you can leave `_get_username_and_password` here, at least for now - we would prefer other modules use more modern and secure forms of logging in when possible.

However, please do modify the configuration to use the astroquery.cfg file instead of a separate config file.

Reply:
Ah, I see that was done in 137eef3, so we're good here.